packages/core/src/components/forms/_control-group.scss

```diff
@@ -66,6 +102,68 @@ Weight: 4
 Styleguide components.forms.control-group
 */
 
+//
+// for best visual results, button group elements should be stacked in the
+// following order to ensure sharp edges in all cases and states:
+//
+//   (lowest z-index)
+//
+//   input fields
+//      :disabled
+//      default
+//
+//   regular buttons / pt-select > select
+//     :disabled, .pt-disabled
+//     default
+//     :focus
+//     :hover
+//     :active, .pt-active
+//
+//   pt-input intent
+//
+//   intent buttons (selects can't be intented yet, but if they could be, they'd be here too)
+//     :disabled, .pt-disabled
+//     default
+//     :focus
+//     :hover
+//     :active, .pt-active
+//
+//   input fields (NOT readonly, disabled, pt-disabled)
+//      :focus
+//
+//   intent input fields (NOT readonly, disabled, pt-disabled)
+//      :focus
+//
+//   input-group children
+//
+//   .pt-select::after (the caret)
+//
+//   (highest z-index)
+
+$control-group-z-index-input-disabled: 1;
```

Comment:
- Yep, just change the z-indexes! 

- Moving everything into a list so we can just use the `index()` helper.